### DISCUSSION TITLE
Husky 9 hooks not working in Gitkraken

### DISCUSSION THREAD

**Discussion by posti85** [typicode/husky#1421] [Q&A] [answered]
husky@9.0.11
Gitkraken 9.13.0
Windows

I have configured a pre-commit hook in my project with husky. And it works fine commiting through a system terminal. However, when I try to commit through Gitkraken UI, the commit is rejected and its console reports:
```
C:\path\to\project\.husk\_\pre-commit:
line 2: C:\path\to\project\.husk\_\pre-commit/h: Not a directory
Hook process exited.
```
I know it could be an issue that has nothing to do with husky, but it worked before when I used husky 8 😣.

Thank you very much.

**Comment by posti85** [comment, marked as answer]
For any one with the same error: I have removed node_modules folder and reinstalled the project, and now the error became to:
```
/usr/bin/env: 'bash': No such file or directory
husky - pre-commit script failed (code 127)
husky - command not found in PATH=...
Hook process exited.
```

But this time, it was easy to fix: just configuring in GitKraken to use the external Windows git (instead of the bundled one) in File > Preferences > Experimental > Git Executable.

And now it works!

**Comment by EsoterikStare** [comment]
I'm having the same issue but unfortunately, nuking node_modules didn't get me anywhere. I also nuked the lock file for good measure, after only nuking node_modules with no change, but that didn't make a difference either.

I'm still having the same error log `.husky\_\pre-commit/h: Not a directory\n Hook process exited.` I've also confirmed that this issue seems specific to GitKraken by committing via the Windows `cmd` terminal and everything worked as expected.

- Windows 11 Pro
- GitKraken v9.13.0 (Pro license, if it matters)
  - Changed Settings > Experimental > Git Executable to use the installed version as suggested above, but no difference.
- Git v2.41.0.windows.1
- husky v9.0.11
- node v20.12.2
- npm v10.5.2

Let me know if I missed any relevant environment info.

Also, I realize this is most likely a GitKraken issue, and I've reported this to their support team, but figured it might be worth documenting here as well for posterity.

Thanks!

**Reply by EsoterikStare** [reply]
I figured out what was going on for me.

I'm not sure how it happened, but somehow, the Git Hooks directory configured in GitKraken `Settings > Git Hooks > Git Hooks directory` was set to `.\husky\_`.

That seemed odd to me, especially after reading the [GitKraken documentation around this](https://help.gitkraken.com/gitkraken-client/githooks/#where-are-git-hooks), so I changed it to `.git\hooks` just to see what would happen and it seems to work normally now...

Hopefully, this helps someone else. 🤷

**Reply by berkon** [reply]
@EsoterikStare, but doing it that way would disable your Husky hooks. My entry looked looked like this: `.husky\_` (I don't know why, looks like sometimes this directory entry gets messed up somehow). But this seems to be wrong. `'_'` is just an internal Husky directory. The Husky hooks are one level up. So after changing the Husky directory to `.husky` in the settings, it started working.

**Reply by wctiger** [reply]
When husky is run from a place like the post installation via "npm prepare" it will recreate the hooks directory: See https://github.com/typicode/husky/blob/da40de3edd1eab39ff28193f748a4efbb05d2d70/index.mjs#L13

I am having the same issue on Windows 10 with GitKraken 10.0.2 and the current workaround is to manually reset the hooksPath back to .husky after installation (It can be automated by a post install script but the idea is the same). Raised a ticket with GitKraken support, will see what they say about it

**Reply by wctiger** [reply]
I was able to fix this by turning on "git executable" from the experimental tab in GitKraken preference and by selecting the git executable installed in Windows. Using the one bundled with GitKraken also worked for me but they also stated in the app that the bundled exe is a minimal version and may be missing some hooks etc. (I only had a simple pre-commit hook and didn't need any advanced hooks)

**Reply by navatweb** [reply]
> I figured out what was going on for me.
> 
> I'm not sure how it happened, but somehow, the Git Hooks directory configured in GitKraken `Settings > Git Hooks > Git Hooks directory` was set to `.\husky\_`.
> 
> That seemed odd to me, especially after reading the [GitKraken documentation around this](https://help.gitkraken.com/gitkraken-client/githooks/#where-are-git-hooks), so I changed it to `.git\hooks` just to see what would happen and it seems to work normally now...
> 
> Hopefully, this helps someone else. 🤷

Just to report that this worked for me too. Thanks.
